pkg/cache/tas_flavor_snapshot.go

```diff
@@ -695,3 +720,19 @@ func (s *TASFlavorSnapshot) notFitMessage(fitCount, totalCount int32) string {
 	}
 	return fmt.Sprintf("topology %q allows to fit only %v out of %v pod(s)", s.topologyName, fitCount, totalCount)
 }
+
+// compileNodeSelector validates and compiles node selectors, returning a selector.
+// Returns labels.Everything() if no selectors are provided.
+// Returns labels.Nothing() if selectors are invalid.
+func (s *TASFlavorSnapshot) compileNodeSelector(nodeSelectors map[string]string) labels.Selector {
+	// If there are no node selectors match everything.
+	if len(nodeSelectors) == 0 {
+		return labels.Everything()
+	}
```

Comment:
```suggestion
	// If there are no node selectors match everything.
	if len(nodeSelectors) == 0 {
		return labels.Everything()
	}
```
I believe this is what happens underneath `labels.ValidateSelectorFromSet` function, according to the documentation https://pkg.go.dev/k8s.io/apimachinery/pkg/labels#ValidatedSelectorFromSet

Unless we want to be verbose about it, I think this might be duplicate